src/main/java/io/cryostat/net/web/http/api/v1/TargetRecordingsPostHandler.java

```diff
@@ -150,18 +131,11 @@ public void handleAuthenticated(RoutingContext ctx) throws Exception {
         }
 
         try {
-            Optional<HyperlinkedSerializableRecordingDescriptor> descriptor =
+            ConnectionDescriptor connectionDescriptor = getConnectionDescriptorFromContext(ctx);
+            HyperlinkedSerializableRecordingDescriptor linkedDescriptor =
```

Comment:
This is for my own understanding; if this `HyperlinkedSerializableRecordingDescriptor ` instance is no longer `Optional<>` does that mean the `linkedDescriptor` can never be `null`? What implications does that have on the code flow, if any?

Reply:
It doesn't guarantee that the `descriptor` is non-null, but it does signify that it isn't expected to be. If `getConnectionDescriptorFromContext` returns null then the `targetConnectionManager.executeConnectedTask` call will fail on the null descriptor, resulting in a NullPointerException that is eventually turned into an HTTP 500 response. The same thing would happen if the `descriptor` is non-null but contains a null `targetId` somehow (which shouldn't happen in practice - how/why would the path parameter be a null reference?).

The reason this simplification was done is because the implementation lower down used to make the call to create the recording, then query for it back (`getRecordingByName`). This was always expected to return back the descriptor for the just-created recording, but it returns an `Optional` since it's generally possible that the given name doesn't match any existing recording. In this refactor, `getRecordingByName` is no longer used - `startRecording` always returned the `descriptor` we needed, so we may as well just use that instead of re-querying for it immediately afterward.